iXBRLViewerPlugin/viewer/src/js/textblockviewer.js

```diff
@@ -0,0 +1,47 @@
+// Copyright 2023 Workiva Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import { Dialog } from './dialog.js';
+
+export class TextBlockViewerDialog extends Dialog {
+    constructor() {
+        super(".dialog.text-block-viewer");
+        this.addButton("Dismiss", true);
+    }
+
+    htmlWrapString(content) {
+        const str =
+            "<!DOCTYPE html><html><head><title></title></head><body>" 
+            + content 
+            + "</body></html>";
+        return str;
+    }
+
+    displayTextBlock(textBlockValue) {
+        const iframe = this.node.find("iframe").get(0);
+        const doc = iframe.contentDocument || iframe.contentWindow.document;
+        doc.open();
+        const html = this.htmlWrapString(textBlockValue);
+        const parser = new DOMParser();
+        const xml = parser.parseFromString(html, 'text/xml');
+        const errorNode = xml.querySelector("parsererror");
+        if (errorNode) {
+            doc.write(this.htmlWrapString('<div style="color: red">Text block contains invalid XML.</div>'));
+        }
+        else {
+            doc.write(html);
```

Comment:
I think it would be helpful to have a comment here explaining:
1. This is safe from an infosec perspective because this code path is only used for escaped=true, which means that this is ultimately just writing a subset of the document content that is already being rendered by the browser.  A comment in `Inspector.prototype._updateValue` might also be welcome.  A method in this class like `TextBlockViewerDialog.canRender(item)` would help so that this potentially unsafe rendering logic and the logic for ensuring it is always safe are colocated.
1. We rely on Arelle to write values in the HTML/XHTML subset so that `<br/>` and `<div></div>` are handled correctly.